File: src/main/java/org/goafabric/core/data/repository/entity/PatientEo.java
package org.goafabric.core.data.repository.entity;

import jakarta.persistence.*;
import org.goafabric.core.data.repository.extensions.AuditListener;

import java.time.LocalDate;

@Entity
@Table(name = "patient")
@EntityListeners(AuditListener.class)
public class PatientEo {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    public String id;

    //@TenantId
    public String orgunitId;

    public String givenName;

    public String familyName;

    public String gender;

    public LocalDate birthDate;

    @OneToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id", referencedColumnName = "id")
    public AddressEo address;

    @OneToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "contact_point_id", referencedColumnName = "id")
    public ContactPointEo contactPoint;

    @Version //optimistic locking
    public Long version;

}

File: src/main/kotlin/org/goafabric/core/organization/persistence/entity/PatientEo.kt
package org.goafabric.core.organization.persistence.entity

import jakarta.persistence.*
import org.goafabric.core.persistence.extensions.AuditTrailListener
import org.goafabric.core.persistence.extensions.KafkaPublisher
import org.hibernate.annotations.TenantId
import java.time.LocalDate

@Entity
@Table(name = "patient")
@EntityListeners(AuditTrailListener::class, KafkaPublisher::class)
class PatientEo(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: String? = null,

    @TenantId
    var organizationId: String? = null,

    var givenName: String? = null,
    var givenSoundex: String? = null,
    var familyName: String? = null,
    var familySoundex: String? = null,
    var gender: String? = null,
    var birthDate: LocalDate? = null,

    @OneToMany(cascade = [CascadeType.ALL])
    @JoinColumn(name = "patient_id")
    var address: MutableList<AddressEo> = mutableListOf(),

    @OneToMany(cascade = [CascadeType.ALL])
    @JoinColumn(name = "patient_id")
    var contactPoint: MutableList<ContactPointEo> = mutableListOf(),

    @Version
    var version: Long? = null
)
